### DISCUSSION TITLE
Timed out fetching a new connection from the connection pool

### DISCUSSION THREAD

**Discussion by alexandrespain** [prisma/prisma#20885] [Q&A]
### Question

Hi team :spock-hand:
I'm creating a new feature `data_migration` as Prisma does not support it under the box using NodeJs. It was working fine until I tested a big data migration (65k rows) then i get the errors shown below then I went on Prisma docs [here](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#disabling-the-pool-timeout) that explains how increase the timeout. I applied it and I still getting same errors.

`Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 17)`

I tried to break my migration in chunks but still receiving the error.

```typescript
class DataMigration {
  constructor(
    private database = new PrismaClient(),
    private concurrencyLimit = 100
  ) {
    this.database = database
    this.concurrencyLimit = concurrencyLimit
  }

  async processRow(row) {
    try {
      await this.database.table2.create({
        data: {
          // data mapping
        }
      })
    } catch (error) {
      console.error('Error processing row:', row.id, error)
    }
  }

  async processChunksSequentially(chunks) {
    if (chunks.length === 0) {
      return
    }

    const chunk = chunks.shift()
    await Promise.all(chunk)
    await this.processChunksSequentially(chunks)
  }

  async table1Totable2() {
    try {
      const sourceData = await this.database.table1.findMany()

      const processChunks = sourceData.reduce(
        (chunks, row) => {
          if (chunks[chunks.length - 1].length >= this.concurrencyLimit) {
            chunks.push([])
          }
          chunks[chunks.length - 1].push(this.processRow(row))
          return chunks
        },
        [[]]
      )

      await this.processChunksSequentially(processChunks)

      // eslint-disable-next-line no-console
      console.log('Data migration completed.')
    } catch (error) {
      console.error('Migration error:', error)
    } finally {
      await this.database.$disconnect()
    }
  }
}

const runDataMigration = async () => {
  const migration = new DataMigration()
  await migration.table1Totable2()
}

runDataMigration()

``` 

Any suggestion? 

### How to reproduce (optional)

RUN:  npx ts-node file_name.ts

### Expected behavior (optional)

I would love that Prisma had this solution under the box but as it is not possible I would expect run all data migration with no timeout issues.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: macOS
- Database: PostgreSQL
- Node.js version: v18.16.0

- Run `prisma -v` to see your Prisma version and paste it
prisma                  : 4.13.0
@prisma/client          : 4.13.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.13.0-50.1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a
Default Engines Hash    : 1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a
Studio                  : 0.484.0

**Comment by nurul3101** [comment]
Hey @alexandrespain 👋 

Can you try enabling [metrics](https://www.prisma.io/docs/concepts/components/prisma-client/metrics) to get insights for your connection pool?

You should get information regarding the status of your connections, which should be helpful in debugging this.

<img width="678" alt="Screenshot 2023-08-31 at 12 10 22 PM" src="https://github.com/prisma/prisma/assets/13049676/259c270d-8292-4a7e-ac0e-98c4b6285137">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
